src/lib/scheduler/operator_task.cpp

```diff
@@ -91,5 +93,23 @@ void OperatorTask::_on_execute() {
 
     context->rollback();
   }
+
+  // Get rid of temporary tables that are not needed anymore
+  if (_cleanup_temporaries) {
+    for (const auto& weak_predecessor : predecessors()) {
+      const auto predecessor = std::dynamic_pointer_cast<OperatorTask>(weak_predecessor.lock());
+      DebugAssert(predecessor != nullptr, "predecessor of OperatorTask is not an OperatorTask itself");
+      bool previous_operator_still_needed = false;
```

Comment:
minor: auto

Reply:
What's the advantage of that?

Reply:
consistency with the rest of the code base
(also, if we use auto consistently, we make sure we never fall into the uninitialised variable trap, at least for local variables)